qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -397,4 +441,128 @@ private static void doCodeActionsForUndefinedSectionTag(Template template, Diagn
+				String signature = null;
+				JsonObject data = (JsonObject) diagnostic.getData();
+				if (data != null) {
+					signature = data.get(DIAGNOSTIC_DATA_TAG).getAsString();
+					collectAvailableValuesForJavaTypePart(node, template, signature, availableValues);
+					break;
+				}
+			}
+
+			for (String value : availableValues) {
+				if (isSimilar(value, varName)) {
+					CodeAction similarCodeAction = CodeActionFactory.replace("Did you mean '" + value + "'?",
+							rangeValue, value, template.getTextDocument(), diagnostic);
+					codeActions.add(similarCodeAction);
+				}
+			}
+		}
+	}
+
+	private void collectAvailableValuesForObjectPart(Node node, Template template, Collection<String> availableValues) {
+		String projectUri = template.getProjectUri();
+
+		List<ValueResolver> globalResolvers = javaCache.getGlobalVariables(projectUri);
+		for (ValueResolver resolver : globalResolvers) {
+			availableValues.add(resolver.getName());
+		}
+
+		List<String> aliases = template.getChildren().stream() //
+				.filter(n -> n.getKind() == NodeKind.ParameterDeclaration) //
+				.map(n -> ((ParameterDeclaration) n).getAlias()) //
+				.filter(alias -> alias != null) //
+				.collect(Collectors.toList());
+		for (String alias : aliases) {
+			availableValues.add(alias);
+		}
+
+		ExtendedDataModelTemplate dataModel = javaCache.getDataModelTemplate(template).getNow(null);
+		if (dataModel != null) {
+			for (ExtendedDataModelParameter parameter : dataModel.getParameters()) {
+				availableValues.add(parameter.getKey());
+			}
+		}
+
+		Section section = node != null ? node.getParentSection() : null;
+		if (section != null) {
+			List<SectionMetadata> metadatas = section.getMetadata();
+			for (SectionMetadata metadata : metadatas) {
+				availableValues.add(metadata.getName());
+			}
+		}
+
+		Collection<SectionMetadata> specialKeysMetadatas = UserTagUtils.getSpecialKeys();
+		for (SectionMetadata metadata : specialKeysMetadatas) {
+			String name = metadata.getName();
+			if (!availableValues.contains(name)) {
+				availableValues.add(name);
+			}
+		}
+	}
+
+	private void collectAvailableValuesForNamespacePart(Node node, Template template,
+			Collection<String> availableValues) {
+		String projectUri = template.getProjectUri();
+
+		String namespace = ((Part) node).getNamespace();
+		if (namespace != null) {
+			List<ValueResolver> namespaceResolvers = javaCache.getNamespaceResolvers(namespace, projectUri);
+			for (ValueResolver resolver : namespaceResolvers) {
+				boolean useNamespaceInTextEdit = namespace == null;
+				String named = resolver.getNamed();
+				if (named != null) {
+					// @Named("user")
+					// User getUser();
+					String label = useNamespaceInTextEdit ? resolver.getNamespace() + ':' + named : named;
+					if (!availableValues.contains(label)) {
+						availableValues.add(label);
+					}
+				}
+			}
+		}
+	}
+
+	private void collectAvailableValuesForJavaTypePart(Node node, Template template, String signature,

```

Comment:
Please remove this code which support similar value for method. We will see that in an another PR.